http4k-format/kondor-json/src/main/kotlin/org/http4k/format/KondorJson.kt

```diff
@@ -0,0 +1,231 @@
+package org.http4k.format
+
+import com.ubertob.kondor.json.JInstance
+import com.ubertob.kondor.json.JsonConverter
+import com.ubertob.kondor.json.jsonnode.JsonNode
+import com.ubertob.kondor.json.jsonnode.JsonNodeArray
+import com.ubertob.kondor.json.jsonnode.JsonNodeBoolean
+import com.ubertob.kondor.json.jsonnode.JsonNodeNull
+import com.ubertob.kondor.json.jsonnode.JsonNodeNumber
+import com.ubertob.kondor.json.jsonnode.JsonNodeObject
+import com.ubertob.kondor.json.jsonnode.JsonNodeString
+import com.ubertob.kondor.json.jsonnode.NodePath
+import com.ubertob.kondor.json.jsonnode.NodePathRoot
+import com.ubertob.kondor.json.jsonnode.NodePathSegment
+import com.ubertob.kondor.json.jsonnode.parseJsonNode
+import com.ubertob.kondor.json.parser.pretty
+import org.http4k.core.Body
+import org.http4k.core.ContentType
+import org.http4k.lens.ContentNegotiation
+import org.http4k.lens.string
+import org.http4k.websocket.WsMessage
+import java.io.InputStream
+import java.math.BigDecimal
+import java.math.BigInteger
+import kotlin.reflect.KClass
+
+class KondorJson(
+    val defaultContentType: ContentType = ContentType.APPLICATION_JSON
+) : AutoMarshallingJson<JsonNode>() {
+
+    private val converters = mutableListOf<ConverterWrapper<*, *>>()
+
+    init {
+        register(Unit::class.java, JInstance(Unit))
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    fun <T : Any> converterFor(target: KClass<T>): ConverterWrapper<T, *> =
+        requireNotNull(converters.find { it.canConvert(target) } as? ConverterWrapper<T, *>) {
+            "JsonConverter for '$target' has not been registered"
+        }
+
+    private fun converterFor(input: Any): ConverterWrapper<*, *> =
+        requireNotNull(converters.find { it.canConvert(input) }) {
+            "JsonConverter for '${input::class}' has not been registered"
+        }
+
+    fun <T: Any, JN : JsonNode> register(target: KClass<T>, converter: JsonConverter<T, JN>): KondorJson = apply {
```

Comment:
In all of the other JSON modules, all of the converters are registered at construction time. Can we please make it match that pattern so we have consistency? 😄

Reply:
No problem, will do that. Will probably mean I can hide some methods/classes that I has to expose publicly to make it work, which is a Good Thing :tm:

Reply:
Done. I'm pretty happy with the outcome - much cleaner now and consistent with other format modules.